airbyte-integrations/bases/base-normalization/integration_tests/test_ephemeral.py

```diff
@@ -0,0 +1,186 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+import json
+import os
+import pathlib
+import shutil
+import tempfile
+from typing import Any, Dict
+
+import pytest
+from integration_tests.dbt_integration_test import DbtIntegrationTest
+from normalization.destination_type import DestinationType
+from normalization.transform_catalog.catalog_processor import CatalogProcessor
+
+temporary_folders = set()
+dbt_test_utils = DbtIntegrationTest()
+
+
+@pytest.fixture(scope="module", autouse=True)
+def before_all_tests(request):
+    dbt_test_utils.change_current_test_dir(request)
+    dbt_test_utils.setup_postgres_db()
+    os.environ["PATH"] = os.path.abspath("../.venv/bin/") + ":" + os.environ["PATH"]
+    yield
+    dbt_test_utils.tear_down_postgres_db()
+    for folder in temporary_folders:
+        print(f"Deleting temporary test folder {folder}")
+        shutil.rmtree(folder, ignore_errors=True)
+
+
+@pytest.fixture
+def setup_test_path(request):
+    dbt_test_utils.change_current_test_dir(request)
+    print(f"Running from: {pathlib.Path().absolute()}")
+    print(f"Current PATH is: {os.environ['PATH']}")
+    yield
+    os.chdir(request.config.invocation_dir)
+
+
+@pytest.mark.parametrize("column_count", [480, 490, 1000])
+@pytest.mark.parametrize(
+    "integration_type",
+    [
+        "Postgres",
+    ],
+)
+# Databases are ran in local containers, we can test more parameters as it's faster
+def test_databases(integration_type: str, column_count: int, setup_test_path):
+    run_test(integration_type, column_count)
+
+
+@pytest.mark.parametrize("column_count", [1000])
+@pytest.mark.parametrize(
+    "integration_type",
+    [
+        "BigQuery",
+        "Snowflake",
+        "Redshift",
+    ],
+)
+def test_warehouses(integration_type: str, column_count: int, setup_test_path):
+    run_test(integration_type, column_count)
```

Comment:
question for understanding: we know this is working as normalisation succeeds even though there are more than 490 columns?

Reply:
yes!

I originally used this test to find out that the limit was at around 490 columns

So without the "fix" avoiding CTEs, it will break normalization with 1000 columns but with the fix it works.

As a side note, normalization still breaks at some point with 1600 something columns anyway because of some other column limits I think...